src/test/java/info/sjd/controller/CartControllerTest.java

```diff
@@ -0,0 +1,235 @@
+        ResponseEntity<Cart> responseEntity = testRestTemplate.exchange(requestEntity, Cart.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(cartService, times(1)).update(any(Cart.class));
+    }
+
+    @Test
+    void updateUnsuccessfulTest() throws URISyntaxException {
+        when(cartService.update(any(Cart.class))).thenReturn(null);
+        RequestEntity<Cart> requestEntity = new RequestEntity<>(new Cart(), HttpMethod.POST, new URI("/cart"));
+        ResponseEntity<Cart> responseEntity = testRestTemplate.exchange(requestEntity, Cart.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.NOT_ACCEPTABLE);
+        verify(cartService, times(1)).update(any(Cart.class));
+    }
+
+    @Test
+    void getByIdTest() throws URISyntaxException {
+        User user = User.builder().
+                login("test_login111").
+                password("test_pass").
+                firstName("test_fn").
+                lastName("test_ln").
+                build();
+        Cart cart = Cart.builder().
+                closed(0).user(user).
+                creationTime(currentTime).
+                build();
+        when(cartService.findById(anyInt())).thenReturn(cart);
+        RequestEntity<Cart> requestEntity = new RequestEntity<>(cart, HttpMethod.GET, new URI("/cart/3"));
+        ResponseEntity<Cart> responseEntity = testRestTemplate.exchange(requestEntity, Cart.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(cartService, times(1)).findById(anyInt());
+    }
+
+    @Test
+    void getAllTest() throws URISyntaxException {
+        User user = User.builder().
+                login("test_login111").
+                password("test_pass").
+                firstName("test_fn").
+                lastName("test_ln").
+                build();
+        Cart cart = Cart.builder().
+                closed(0).user(user).
+                creationTime(currentTime).
+                build();
+        when(cartService.findAll()).thenReturn(Collections.singletonList(cart));
+        RequestEntity<Cart> requestEntity = new RequestEntity<>(new Cart(), HttpMethod.GET, new URI("/cart"));
+        ResponseEntity<List> responseEntity = testRestTemplate.exchange(requestEntity, List.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(cartService, times(1)).findAll();
+    }
+
+    @Test
+    void deleteTest() throws URISyntaxException {
+        User user = User.builder().
+                login("test_login111").
+                password("test_pass").
+                firstName("test_fn").
+                lastName("test_ln").
+                build();
+        Cart cart = Cart.builder().
+                closed(0).user(user).
+                creationTime(currentTime).
+                build();
+        when(cartService.findById(anyInt())).thenReturn(cart);
```

Comment:
You must mock delete method too. 
Like
doNothing().when(cartService)...

Reply:
done